onnxruntime/core/platform/windows/env.cc

```diff
@@ -141,6 +141,30 @@ class WindowsEnv : public Env {
     return processorCoreCount;
   }
 
+  std::vector<size_t> GetThreadAffinityMasks() const override {
+    auto generate_vector_of_n = [](int n) {
+      std::vector<size_t> ret(n);
+      std::iota(ret.begin(), ret.end(), 0);
+      return ret;
+    };
+    // Indeed 64 should be enough. However, it's harmless to have a little more.
+    SYSTEM_LOGICAL_PROCESSOR_INFORMATION buffer[256];
+    DWORD returnLength = sizeof(buffer);
+    if (GetLogicalProcessorInformation(buffer, &returnLength) == FALSE) {
+      return generate_vector_of_n(std::thread::hardware_concurrency());
+    }
+    std::vector<size_t> ret;
+    int count = (int)(returnLength / sizeof(SYSTEM_LOGICAL_PROCESSOR_INFORMATION));
+    for (int i = 0; i != count; ++i) {
+      if (buffer[i].Relationship == RelationProcessorCore) {
+        ret.push_back(buffer[i].ProcessorMask);
```

Comment:
Note that this code path let each thread bind with a physical core (instead of logicial core). The behavior is different with the that in code path of generate_vector_of_n (which binds thread with logical cores) when hyperthreading is enabled.

Reply:
The  generate_vector_of_n function call is a fallback path. If we couldn't get the information of physical cores, then bind it to every logical core. So I think it's not a mismatch.

Reply:
Added a task https://msdata.visualstudio.com/Vienna/_workitems/edit/722456